src/features/environment.js

```diff
@@ -43,9 +45,22 @@ export const EnvironmentMixin = (ModelViewerElement) => {
       this[$envMapGenerator] = new EnvMapGenerator(this[$renderer].renderer);
     }
 
+    get [$hasBackgroundImage]() {
+      // @TODO #76
+      return this.backgroundImage && this.backgroundImage !== 'null';
+    }
+
+    get [$hasBackgroundColor]() {
+      // @TODO #76
+      return this.backgroundColor && this.backgroundColor !== 'null';
+    }
+
     connectedCallback() {
       super.connectedCallback();
-      this[$setEnvironmentColor]();
+
+      if (!this[$hasBackgroundImage] && !this[$hasBackgroundColor]) {
+        this[$setEnvironmentColor]();
```

Comment:
Two things:

 1. Let's do this check in `update`. `update` is _always_ called at least once (and will be called at `connectedCallback` if not due to attribute change).
 2. I think the method invocation would be easier to read if it didn't rely on implicit side-effects of the implementation. Let's invoke it explicitly with the default color `this[$setEnvironmentColor](DEFAULT_BACKGROUND_COLOR)`, or maybe imagine a different name for the method.

Reply:
:+1: